### PR TITLE
SCT-815 update mongodb record with new mosaic id

### PR DESCRIPTION
**Pull request by LBHTKarki**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/SCT-815

### *What is the problem we're trying to solve*

When person records are soft deleted in postgreSQL the existing form records must be transferred over to the new master person id. This scripts updates all relevant records for a given ID in `form_data` collection.

Comments from the script:
**summary**
 This script is designed to be run against a single ID as a workaround until the appropriate API endpoint is in place
 Only runs against form_data collection
 Updates all matching records that are not any of the audit event types created by the service API

**Howto**
To run this script manually on the server:
- remove double quotes
- add appropriate IDs (one in the query (q) is the current one and the one in the $set is the new master id)
- pass it to the db.runCommand() like this:

```
db.runCommand({
 update: "form_data",
 updates:
     [
         {
              q: { MosaicId: "334445566", FormNameOverall: { $not: /^API_/ } }, u: { $set: { MosaicId: "77889900" } }, multi: true
         }
     ]
});
```

### *What changes have we introduced*

Added a script that can be run against form_data collection to update the form records.
Please note this does not handle any other collections, so residents submissions collection for example has to be handled separately (done in separate PR once the transfer method is agreed).

Re-structured data script tests to new namsespaces

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        [Test]
+        public void DoesNotUpdateOtherThanMatchingRecords()
+        {
+            string mosaicId = "334445566";
+            string mosaicIdTwo = "556677";
+            string masterPersonId = "77889900";
+
+            var casenoteOne = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).Create();
+            var casenoteTwo = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).Create();
+            var casenoteThree = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicIdTwo).Create();
```
Comment (chubberlisk) [inline SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs:109]:
```suggestion
            var casenoteOneForMosaicId = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).Create();
            var casenoteTwoForMosaicId = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).Create();
            var casenoteThreeForMosaicIdTwo = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicIdTwo).Create();
```

Or something like that to make it clear that the third case note is for another person.

File: SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+            collection?.InsertOne(casenoteOne);
+            collection?.InsertOne(casenoteTwo);
+            collection?.InsertOne(casenoteThree);
+
+            var result = _mongoDatabase?.RunCommand(_dbCommandUnderTest);
+
+            var filterByNewId = Builders<MongoDBTestObject>.Filter.Eq("MosaicId", masterPersonId);
+            var recordsWithNewId = collection.Find(filterByNewId).ToList();
+            recordsWithNewId.Count.Should().Be(2);
```
Comment (chubberlisk) [inline SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs:148]: What do you think of changing the expectation to find the case note with the `FormNameOverall` is `API_Event` and checking that the `MosaicId` is still `mosaicId`? I think that's more precise than checking the count.

**Review by chubberlisk — APPROVED** [review]

File: SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs (hunk 1 above)
Comment (LBHTKarki) [reply SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs]: @chubberlisk I've renamed them. That's better

File: SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs (hunk 2 above)
Comment (LBHTKarki) [reply SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs]: @chubberlisk That's more robust, I've pushed an update.

**Pull request merged**